apps/server/src/modules/h5p-library-management/service/h5p-library-management.service.spec.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Test, TestingModule } from '@nestjs/testing';
+import { DeepMocked, createMock } from '@golevelup/ts-jest';
+import { ContentStorage, LibraryStorage } from '@src/modules/h5p-editor/service';
+import { LibraryAdministration, ContentTypeCache } from '@lumieducation/h5p-server';
+import { IHubContentType } from '@lumieducation/h5p-server/build/src/types';
+import { s3ConfigContent, s3ConfigLibraries } from '@src/modules/h5p-editor/h5p-editor.config';
+import { H5PLibraryManagementService } from './h5p-library-management.service';
+
+jest.mock('@lumieducation/h5p-server', () => {
+	return {
+		...jest.requireActual('@lumieducation/h5p-server'),
+		LibraryAdministration: jest.fn().mockImplementation(() => {
+			return {
+				getLibraries: jest.fn().mockResolvedValue([
+					{ machineName: 'a', dependentsCount: 0 },
+					{ machineName: 'b', dependentsCount: 1 },
+					{ machineName: 'c', dependentsCount: 0 },
+				]),
+			};
+		}),
+	};
+});
+
+describe('H5PLibraryManagementService', () => {
+	let module: TestingModule;
+	let service: H5PLibraryManagementService;
+	let libraryStorageMock: DeepMocked<LibraryStorage>;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				H5PLibraryManagementService,
+				{
+					provide: LibraryStorage,
+					useValue: createMock<LibraryStorage>(),
+				},
+				{
+					provide: ContentStorage,
+					useValue: createMock<ContentStorage>(),
+				},
+			],
+		}).compile();
+
+		service = module.get(H5PLibraryManagementService);
+		libraryStorageMock = module.get(LibraryStorage);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('uninstallUnwantedLibraries', () => {
+		it('should delete libraries not in the wanted list and with no dependents', async () => {
+			const s3ConfigLibrariess = s3ConfigLibraries;
+			libraryStorageMock.deleteLibrary = jest.fn().mockResolvedValue({});
```

Comment:
It is already mocked over golevelup. You do not need jest for it. 
libraryStorageMock.deleteLibrary.mockResolvedValue
mockResolvedValue is for syncron operations. MockResolveValue and mockRejectValue for async. 
You are sure that you use the corect methode?

We also always use the Once. MockResolveValueOnce() to make saver to use.

Reply:
solved: c8a5a2795d682749dc7cff512e1123b51c1526d3